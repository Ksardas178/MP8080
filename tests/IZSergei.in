LXI B o200 o000 //Задаем начальный адрес
MVI D 10 //Инициализируем счетчик
MVI E 0 //Переменная под максимальное число

>LABEL
	/*TODO*/
	LDAX B 	//A <- arr[i]
	
	SUB E	//A <- A - E
	JM 14 0//if (A < 0) goto ENDIF
		//else store new max element
		LDAX B	//A <- arr[i]
		MOV E A	//E <- A*
	>ENDIF

	INX B	//i++	
	DCR D 	//D--
JNZ LABEL 	//if (D != 0) goto LABEL

/*Store max value to addr*/	
MVI A E
STA o100 o000

HLT