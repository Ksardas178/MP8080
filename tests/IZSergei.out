stored label LABEL on line 7
stored label ENDIF on line 14

[32;1mInput is correct
[0m
[30;1mCode analysis results:
[0m______________________________
00000000: 00000001
00000001: 10000000
00000010: 00000000
00000011: 00010110
00000100: 00001010
00000101: 00011110
00000110: 00000000
00000111: 00001010
00001000: 10010011
00001001: 11111010
00001010: 00001110
00001011: 00000000
00001100: 00001010
00001101: 01011111
00001110: 00000011
00001111: 00010101
00010000: 11000010
00010001: 00000111
00010010: 00000000
00010011: 00111110
00010100: 00000011
00010101: 00110010
00010110: 01000000
00010111: 00000000
00011000: 01110110
